src/library/Server/Manager/Plesk.php

```diff
@@ -84,452 +134,613 @@ public function createAccount(Server_Account $account)
-                        ],
-                        [
+                            'value' => $package->getMaxFtp() ? 'true' : 'false',
+                        ),
+                        array(
                             'name' => 'manage_spamfilter',
-                            'value' => $p->getHasSpamFilter() ? 'true' : 'false',
-                        ],
-                    ],
-                ],
-            ],
-        ];
+                            'value' => $package->getHasSpamFilter() ? 'true' : 'false',
+                        ),
+                    ),
+                ),
+            ),
+        );
     }
 
     /**
-     * Creates a new client account.
+     * Modifies the properties of a client account on the Plesk server.
      *
-     * @return int client's Plesk id
+     * @param Server_Account $account The account for which the properties should be modified.
+     * @return mixed The result of the API request to modify the client's properties.
+     */
+    private function modifyClient(Server_Account $account): mixed
+    {
+        if ($account->getReseller()) {
+            $result = $this->_client->reseller()->setProperties('login', $account->getUsername(), $this->createClientProps($account));
+        } else {
+            $result = $this->_client->customer()->setProperties('login', $account->getUsername(), $this->createClientProps($account));
+        }
+
+        return $result;
+    }
+
+    /**
+     * Creates an array of properties for a client account.
+     * The properties include the client's company name, full name, username, password, telephone number, fax number, email address, address, city, and state.
      *
-     * @throws Server_Exception
+     * @param Server_Account $account The account for which the properties should be created.
+     * @return array The array of client properties.
      */
-    private function _createClient(Server_Account $a)
+    private function createClientProps(Server_Account $account): array
     {
-        $client = $a->getClient();
+        $client = $account->getClient();
 
-        $props = [
+        return [
             'cname' => $client->getCompany(),
             'pname' => $client->getFullname(),
-            'login' => $a->getUsername(),
-            'passwd' => $a->getPassword(),
+            'login' => $account->getUsername(),
+            'passwd' => $account->getPassword(),
             'phone' => $client->getTelephone(),
             'fax' => $client->getFax(),
             'email' => $client->getEmail(),
             'address' => $client->getAddress1(),
             'city' => $client->getCity(),
             'state' => $client->getState(),
-            'description' => 'Created using FOSSBilling.',
         ];
+    }
 
-        if ($a->getReseller()) {
-            $result = $this->_client->reseller()->create($props);
-        } else {
-            $result = $this->_client->customer()->create($props);
+    /**
```

Comment:
Looks like these aren't supposed to be part of this PR, please remove them

Reply:
Oh, It looks like I accidentally un-highlighted the lines in question.
It's supposed to be surrounding the DNS related functions you added which appear to be part of something else you are working on which is separate from this PR

Reply:
Hmm, which DNS functions are you referring to? There's a chance things ended up getting merged that weren't supposed to, but I'm not sure I see anything my end

Reply:
Nevermind.. I realize now they were already there and had just been renamed, but I didn't notice that because the file had been shifted around and the comparison view was a bit off

Reply:
Ah. Yeah, I'm still dialing in my IDE's configuration for PHP, which probably hasn't helped